### PR TITLE
Added Atom feed generation functionality #6976

### PR DESCRIPTION
**Pull request by PritishC**
Cherry-picked for v3.4.

### CODE REVIEW

File: cms.py [hunk 1]
```diff
@@ -698,6 +734,15 @@ def default_content_type(cls):
         """
         return 'plain'
 
+    @classmethod
+    def get_published_articles(cls):
```
Comment (tarunbhardwaj) [inline cms.py]: need not to create new method

File: cms.py [hunk 2]
```diff
@@ -846,6 +891,68 @@ def get_menu_item(self, max_depth):
...
+    @classmethod
+    @route('/article/all.atom')
+    def atom_feed(cls):
+        """
+        Renders the atom feed for all articles.
+        """
+        feed = AtomFeed(
+            "All Articles", feed_url=request.url, url=request.host_url
+        )
+        for article_id in cls.get_published_articles():
```
Comment (tarunbhardwaj) [inline cms.py]: search directly

File: cms.py [hunk 3]
```diff
@@ -601,6 +604,39 @@ def get_children(self, max_depth):
...
         ]
 
+    def serialize(self, purpose=None):
+        """
+        Article category serialize method
+        """
+        if purpose == 'atom':
+            return {
+                'term': self.unique_name,
+            }
```
Comment (tarunbhardwaj) [inline cms.py]: else if super hasattr serialize call that and return

File: cms.py [hunk 4]
```diff
@@ -846,6 +891,68 @@ def get_menu_item(self, max_depth):
...
+                    'type': 'text/html',
+                    'href': self.get_absolute_url(external=True),
+                },
+                'category': [
+                    category.serialize(purpose=purpose)
+                    for category in self.categories
+                ],
+                'published': self.atom_publish_date(),
+                'updated': self.write_date or self.atom_publish_date(),
+            }
```
Comment (tarunbhardwaj) [inline cms.py]: else if super hasattr serialize call that and return

File: user.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    def serialize(self, purpose=None):
+        """
+        Downstream implementation of serialize() which adds serialization for
+        atom feeds.
+        """
+        if purpose == 'atom':
+            return {
+                'name': self.display_name,
+                'email': self.email or None,
+            }
```
Comment (tarunbhardwaj) [inline user.py]: else if super hasattr serialize call that and return

File: user.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        try:
+            author = cls(id)
+        except:
+            abort(404)
+
+        feed = AtomFeed(
+            "Articles by Author %s" % author.display_name, feed_url=request.url,
+            url=request.host_url
+        )
+        for article_id in author.get_published_articles():
```
Comment (tarunbhardwaj) [inline user.py]: search directly

File: cms.py (hunk 3 above)
Comment (PritishC) [inline cms.py]: Done

File: cms.py (hunk 4 above)
Comment (PritishC) [inline cms.py]: Done

File: user.py (hunk 5 above)
Comment (PritishC) [inline user.py]: Done

File: cms.py [hunk 7]
```diff
@@ -601,6 +604,41 @@ def get_children(self, max_depth):
...
+                ('unique_name', '=', uri),
+            ], limit=1)
+        except ValueError:
+            abort(404)
+
+        feed = AtomFeed(
+            "Articles by Category %s" % category.unique_name,
+            feed_url=request.url, url=request.host_url
+        )
+        for article_id in category.published_articles:
```
Comment (tarunbhardwaj) [inline cms.py]: published_articles will be list of active records only.

**Pull request merged**